src/pages/api/revalidate-tutorial.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+import { validateToken } from 'lib/api-validate-token'
+import {
+	ApiFeaturedCollection,
+	ApiProductsUsed,
+	ApiTutorialLite,
+} from 'lib/learn-client/api/api-types'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+
+/**
+ * Accepts a POST request with a tutorial object, triggers revalidation for all paths associated with that tutorial
+ * along with the parent collection pages and product tutorial landing paths for the products used.
+ */
+async function handler(request: NextApiRequest, response: NextApiResponse) {
+	if (request.method !== 'POST') {
+		response.status(StatusCodes.NOT_FOUND)
+		return
+	}
+
+	const { tutorial }: { tutorial: ApiTutorialLite } = request.body
+
+	if (!tutorial) {
+		response
+			.status(StatusCodes.BAD_REQUEST)
+			.json({ error: '[Revalidation failed]: No tutorial object provided.' })
+
+		return
+	}
+
+	try {
+		const revalidatePromises = []
+		const paths = getPathsToRevalidate(tutorial)
+
+		if (!paths || paths.length === 0) {
+			response.status(200).end()
+		}
+
+		paths.forEach((path: string) => {
+			console.log('[Revalidate]', path)
+			revalidatePromises.push(response.revalidate(path))
+		})
+
+		// TODO: Add resiliency here, with concurrency or batching
+		await Promise.allSettled(revalidatePromises)
+
+		response.status(200).end()
+	} catch (e) {
+		// If there was an error, Next.js will continue
+		// to show the last successfully generated page
+		console.error('Error revalidating ', e)
+		return response.status(500).send('Error revalidating')
+	}
+}
+
+export default validateToken(handler, {
+	token: process.env.REVALIDATE_TOKEN,
+	onlyMethods: ['POST'],
+})
+
+/**
+ * TODO consider busting the cache for all other tutorials in the collections as well
+ * this would only matter in the case of a name change for the sidebar data.
+ *
+ * This would require an additional API call as we don't have the tutorial
+ * array within the featured collection objects with this return data.
+ *
+ * See the `augmentTutorial` function in src/lib/learn-client/api/tutorial/augment-tutorial.ts
+ * for an example of getting full tutorial data from featured collections.
```

Comment:
I'm leaning towards going the 'nuclear' route instead where whenever a pr is merged to main in tutorials repo, we bust the entire cache for all tutorials sub-routes. 

This finite approach is nice in theory but I see a need to build the workflow when we bust the entire cache anyways since this won't get _everything_. Perhaps until we get [this feature](https://vercel.com/blog/vercel-cache-api-nextjs-cache), we could hold off here.